sklearn/metrics/tests/test_pairwise.py

```diff
@@ -146,8 +152,8 @@ def test_pairwise_distances(global_dtype):
 
     # Test with sparse X and Y,
     # currently only supported for Euclidean, L1 and cosine.
-    X_sparse = csr_matrix(X)
-    Y_sparse = csr_matrix(Y)
+    X_sparse = csr_container(X)
```

Comment:
Could we isolate the code that test with the sparse matrices.
It would avoid to repeat the tests for the dense case multiple times.

Reply:
@glemaitre, do you mean to turn this test into two separate tests?

Reply:
Yes exactly.

Reply:
Okay, I will try that. :)